Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -401,7 +406,8 @@ class Gx2Fitter {
       // - Waiting for a current surface
       auto surface = navigator.currentSurface(state.navigation);
       //      std::string direction = state.stepping.navDir.toString();
-      if (surface != nullptr) {
+      if (surface != nullptr &&
+          surface->associatedDetectorElement() != nullptr) {
```

Comment:
why is this additional check necessary?

Reply:
Otherwise we would also go to surfaces without measurements.

Reply:
not sure if that is the right filter for a surface with measurements. you can also land on a surface which does have a detector element attached but without any measurements no?

Reply:
Ehm yes, I meant surfaces, that could contain measurements. @asalzburger suggested that fix

Reply:
below you have a check if the surface has a measurement

```
// check if measurement surface
auto sourcelink_it = inputMeasurements->find(surface->geometryId());
if (sourcelink_it != inputMeasurements->end()) {
  // measurement
}
```

this is what I would do instead of `surface->associatedDetectorElement() != nullptr` if you want to filter for surfaces with measurement at that stage. if so I guess the code can be a bit rearranged to avoid double `find`